pkg/cmd/dev/doctor.go

```diff
@@ -95,8 +96,67 @@ func makeDoctorCmd(runE func(cmd *cobra.Command, args []string) error) *cobra.Co
 
 func (d *dev) doctor(cmd *cobra.Command, _ []string) error {
 	ctx := cmd.Context()
-	success := true
 
+	if err := d.setupGithooks(ctx); err != nil {
+		return err
+	}
+	success, err := d.checkRequiredTools(ctx)
+	if err != nil {
+		return err
+	}
+	if !success {
+		return errors.New("please address the errors described above and try again")
+	}
+	if err := d.writeDoctorStatus(ctx, d.exec); err != nil {
+		return err
+	}
+	log.Println("You are ready to build :)")
+	return nil
+}
+
+func (d *dev) setupGithooks(ctx context.Context) error {
+	hookdirStdout, err := d.exec.CommandContextSilent(ctx, "git", "rev-parse", "--git-path", "hooks")
+	if err != nil {
+		return err
+	}
+	hooksDir, err := filepath.Abs(strings.TrimSpace(string(hookdirStdout)))
+	if err != nil {
+		return err
+	}
+	if err := os.Mkdir(hooksDir, 0700); err != nil && !errors.Is(err, os.ErrExist) {
+		return err
+	}
+
+	dir, err := d.getWorkspace(ctx)
+	if err != nil {
+		return err
+	}
+	srcHooks := filepath.Join(dir, "githooks")
+
+	return filepath.Walk(srcHooks, func(s string, i fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if i.IsDir() {
+			return nil
+		}
+		link := filepath.Join(hooksDir, filepath.Base(s))
+		if _, err := os.Stat(link); err == nil {
+			return nil
+		}
+
+		target, err := filepath.Rel(hooksDir, s)
+		if err != nil {
+			return err
+		}
+		log.Printf("linking %s to %s", link, target)
+		os.Symlink(target, link)
+		return nil
+	})
+}
+
+func (d *dev) checkRequiredTools(ctx context.Context) (bool, error) {
```

Comment:
RE: the type change here -- it looks like the returned `bool` is `true` iff `err != nil`, right? Seems like we can just return the `error` in that case, right?